packages/flutter_tools/lib/src/web/bootstrap.dart

```diff
@@ -4,6 +4,320 @@
+  required String mapperUrl,
+  required bool generateLoadingIndicator,
+}) {
+  return '''
+${generateLoadingIndicator ? _generateLoadingIndicator() : ""}
+$_currentDirectoryScript
+$_simpleLoaderScript
+
+// A map containing the URLs for the bootstrap scripts in debug.
+let _scriptUrls = {
+  "mapper": "$mapperUrl",
+  "moduleLoader": "$ddcModuleLoaderUrl"
+};
+
+// Create a TrustedTypes policy so we can attach Scripts...
+let _ttPolicy;
+if (window.trustedTypes) {
+  _ttPolicy = trustedTypes.createPolicy("flutter-tools-bootstrap", {
+    createScriptURL: (url) => {
+      let scriptUrl = _scriptUrls[url];
+      if (!scriptUrl) {
+        console.error("Unknown Flutter Web bootstrap resource!", url);
+      }
+      return scriptUrl;
+    }
+  });
+}
+
+// Creates a TrustedScriptURL for a given `scriptName`.
+// See `_scriptUrls` and `_ttPolicy` above.
+function getTTScriptUrl(scriptName) {
+  let defaultUrl = _scriptUrls[scriptName];
+  return _ttPolicy ? _ttPolicy.createScriptURL(scriptName) : defaultUrl;
+}
+
+(function() {
+  let appName = "$entrypoint";
+
+  // A uuid that identifies a subapp.
+  // Stubbed out since subapps aren't supported in Flutter.
+  let uuid = "00000000-0000-0000-0000-000000000000";
+
+  window.postMessage(
+      {type: "DDC_STATE_CHANGE", state: "initial_load", targetUuid: uuid}, "*");
+
+  // Load pre-requisite DDC scripts.
+  // We intentionally use invalid names to avoid namespace clashes.
+  let prerequisiteScripts = [
+    {
+      "src": getTTScriptUrl("moduleLoader"),
+      "id": "ddc_module_loader \x00"
+    },
+    {
+      "src": getTTScriptUrl("mapper"),
+      "id": "dart_stack_trace_mapper \x00"
+    }
+  ];
+
+  // Load ddc_module_loader.js to access DDC's module loader API.
+  let prerequisiteLoads = [];
+  for (let i = 0; i < prerequisiteScripts.length; i++) {
+    prerequisiteLoads.push(forceLoadModule(prerequisiteScripts[i].src));
+  }
+  Promise.all(prerequisiteLoads).then((_) => afterPrerequisiteLogic());
+
+  // Save the current script so we can access it in a closure.
+  var _currentScript = document.currentScript;
+
+  var afterPrerequisiteLogic = function() {
+    window.\$dartLoader.rootDirectories.push(_currentDirectory);
+    let scripts = [
+      {
+        "src": "dart_sdk.js",
+        "id": "dart_sdk"
+      },
+      {
+        "src": "main_module.bootstrap.js",
+        "id": "data-main"
+      }
+    ];
+    let loadConfig = new window.\$dartLoader.LoadConfiguration();
+    loadConfig.bootstrapScript = scripts[scripts.length - 1];
+
+    if (window.\$dartJITModules) {
+      loadConfig.loadScriptFn = function(loader) {
+        // Loads just the entrypoint module and required SDK modules.
+        let moduleSet = new Set();
+        // This cache is populated by ddc_module_loader.js
+        let libraryCache = JSON.parse(window.localStorage.getItem(`dartLibraryCache:\${appName}`));
+        if (libraryCache) {
+          // TODO(b/165021238) - when should this be invalidated?
```

Comment:
Can this `b` issue have an external counterpart in github, that is kept in sync with the internal one (so improvements in the DDC loading make it to the open source version as well!)

Reply:
JIT modules isn't supported externally (yet), so this is technically dead code. Removed! 

I'll port the bugs if/when we externalize it.